### ISSUE TITLE
All newly uploaded images are black in gallery?

### ISSUE THREAD

**Issue by gardiol** [LycheeOrg/Lychee#1498]
### Detailed description of the problem [REQUIRED]

Just updated to latest RC (from very old release)

Now, when i upload a bunch of pictures, they show up as just black. I can see the bottom bar when moving on them and i can open them and see individually, but in the gallery just black.

### Steps to reproduce the issue

**Screenshots**

### Output of the diagnostics  [REQUIRED]

    Diagnostics
    -------
    Warning: Dropbox import not working. dropbox_key is empty.
    Warning: zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.
    Info: Latest version of PHP is 8.1
    Warning: Default timezone not properly set; you might experience strange results when importing photos without explicit EXIF timezone

    System Information
    --------------
    Lychee Version (release):        4.6.0
    DB Version:                      4.6.0
    
    composer install:                --no-dev
    APP_ENV:                         local
    APP_DEBUG:                       true
    
    System:                          Linux
    PHP Version:                     8.0.12
    PHP User agent:                  Lychee/4 (https://lycheeorg.github.io/)
    Timezone:                        UTC
    Max uploaded file size:          500M
    Max post size:                   500M
    Max execution time:              200
    MySQL Version:                   10.5.10-MariaDB-log
    
    exec() Available:                yes
    Imagick Available:               1
    Imagick Enabled:                 1
    Imagick Version:                 1808
    GD Version:                      bundled (2.1.0 compatible)

Config Information
    --------------
    version:                         040600
    check_for_updates:               0
    sorting_photos_col:              title
    sorting_photos_order:            ASC
    sorting_albums_col:              title
    sorting_albums_order:            DESC
    imagick:                         1
    skip_duplicates:                 0
    small_max_width:                 0
    small_max_height:                360
    medium_max_width:                1920
    medium_max_height:               1080
    lang:                            it
    layout:                          1
    image_overlay_type:              desc
    default_license:                 reserved
    compression_quality:             90
    full_photo:                      1
    delete_imported:                 0
    Mod_Frame:                       1
    Mod_Frame_refresh:               30
    thumb_2x:                        1
    small_2x:                        1
    medium_2x:                       1
    landing_page_enable:             1
    landing_owner:                   Willy Gardiol
    landing_title:                   Willy Gardiol Photos
    landing_subtitle:                Foto di Willy Gardiol
    landing_facebook:                
    landing_flickr:                  
    landing_twitter:                 
    landing_instagram:               
    landing_youtube:                 
    landing_background:              dist/cat.jpg
    site_title:                      Willy Gardiol Photos
    site_copyright_enable:           1
    site_copyright_begin:            2019
    site_copyright_end:              2019
    additional_footer_text:          
    display_social_in_gallery:       0
    public_search:                   0
    SL_enable:                       0
SL_for_admin:                    0
    public_recent:                   0
    recent_age:                      1
    public_starred:                  0
    downloadable:                    0
    photos_wraparound:               1
    map_display:                     1
    zip64:                           1
    map_display_public:              1
    map_provider:                    Wikimedia
    force_32bit_ids:                 0
    map_include_subalbums:           1
    update_check_every_days:         3
    has_exiftool:                    0
    share_button_visible:            0
    has_ffmpeg:                      0
    import_via_symlink:              0
    location_decoding:               1
    location_decoding_timeout:       30
    location_show:                   1
    location_show_public:            0
    rss_enable:                      0
    rss_recent_days:                 7
    rss_max_items:                   100
    prefer_available_xmp_metadata:   0
    editor_enabled:                  1
    lossless_optimization:           0
    swipe_tolerance_x:               150
    swipe_tolerance_y:               250
    local_takestamp_video_formats:   .avi|.mov
    log_max_num_line:                1000
    unlock_password_photos_with_url_param: 0
    nsfw_visible:                    1
    nsfw_blur:                       0
    nsfw_warning:                    0
    nsfw_warning_admin:              0
    map_display_direction:           1
    album_subtitle_type:             oldstyle
    upload_processing_limit:         4
    public_photos_hidden:            1
    new_photos_notification:         0
    legacy_id_redirection:           1
    zip_deflate_level:               6

**Comment by gardiol** [comment]
I stand corrected: i cannot even see the big image.
In the logs i only have this:
2022-09-01 10:10:26 UTC -- error   -- Illuminate\Routing\AbstractRouteCollection::handleMatchedRoute -- 43 -- 
2022-09-01 10:10:26 UTC -- error   -- App\Http\Middleware\FixStatusCode::handle -- 31 -- ; caused by
2022-09-01 10:10:26 UTC -- error   -- Illuminate\Routing\AbstractRouteCollection::handleMatchedRoute -- 43 -- 
2022-09-01 10:10:26 UTC -- error   -- App\Http\Middleware\FixStatusCode::handle -- 31 -- ; caused by
2022-09-01 10:10:26 UTC -- error   -- Illuminate\Routing\AbstractRouteCollection::handleMatchedRoute -- 43 -- 
2022-09-01 10:10:26 UTC -- error   -- App\Http\Middleware\FixStatusCode::handle -- 31 -- ; caused by
(i cleared them before importing)

**Comment by nagmat84** [comment]
Is this the complete log? The actual error messages (i.e. the message of the exceptions) are missing. Did you remove them before posting it here or this really what you see in the logs?

**Comment by gardiol** [comment]
I just hit the "clear" log before creating the new album and uploading the photos

**Comment by nagmat84** [comment]
That's not what I meant.

For example, you have the following two following log entries:

```
2022-09-01 10:10:26 UTC -- error -- Illuminate\Routing\AbstractRouteCollection::handleMatchedRoute -- 43 -- 
```

and 

```
2022-09-01 10:10:26 UTC -- error -- App\Http\Middleware\FixStatusCode::handle -- 31 -- ; caused by
```

There should be a proper error message after the last double dash `--` for each entry like this:

```
2022-09-01 10:10:26 UTC -- error -- Illuminate\Routing\AbstractRouteCollection::handleMatchedRoute -- 43 -- HERE SHOULD BE AN ERROR MESSAGE 
```

and 

```
2022-09-01 10:10:26 UTC -- error -- App\Http\Middleware\FixStatusCode::handle -- 31 -- HERE SHOULD BE AN ERROR MESSAGE; caused by
```

The log entries _themselves_ are incomplete.

**Comment by gardiol** [comment]
Where should i find the logs, beside the GUI?
that all i see on the GUI

**Comment by nagmat84** [comment]
WTF?! If you have CLI access to your DB, the logs are stored in the table `logs`, hence ```select * from `logs` order by created_at;``` should give you the logs as well. But somehow I suspect that the error messages are missing there, too, which is crazy.

**Comment by nagmat84** [comment]
Btw, some of your settings are weird:

```
composer install: --no-dev
APP_ENV: local
APP_DEBUG: true
```

Enabling debugging without having the debug packages installed makes not so much sense.

Also, there is no application environment `local`. The only valid values are `production` and `developing` (or `development`, I always forget). Please fix these settings in your `.env` file. Although they probably won't fix the problem at hand, they might somehow interfere with the error messages which might be an explanation why we don't see them. (But this is merely a wild guess.) Nonetheless, it won't harm to set the options to proper values.

**Comment by ildyria** [comment]
> Enabling debugging without having the debug packages installed makes not so much sense.

Actually it is possible, if I remember correctly, if you don't put Debug=true, Laravel will just return 500 on error without details. However if you put debug on you will get the stack trace but not the variables on the right.

**Comment by gardiol** [comment]
select * from `logs` order by created_at;


id	created_at   	updated_at	type	function	line	text	
1	2022-09-01 10:10:26.413265	2022-09-01 10:10:26.413265	error	App\Http\Middleware\FixStatusCode::handle	31	; caused by	
2	2022-09-01 10:10:26.421890	2022-09-01 10:10:26.421890	error	Illuminate\Routing\AbstractRouteCollection::handle...	43		
3	2022-09-01 10:10:26.427300	2022-09-01 10:10:26.427300	error	App\Http\Middleware\FixStatusCode::handle	31	; caused by	
4	2022-09-01 10:10:26.433633	2022-09-01 10:10:26.433633	error	Illuminate\Routing\AbstractRouteCollection::handle...	43		
5	2022-09-01 10:10:26.473286	2022-09-01 10:10:26.473286	error	App\Http\Middleware\FixStatusCode::handle	31	; caused by	
6	2022-09-01 10:10:26.479482	2022-09-01 10:10:26.479482	error	Illuminate\Routing\AbstractRouteCollection::handle...	43		

the query results

**Comment by gardiol** [comment]
In the past i did some development (picture rotation stuff) so maybe those .env are leftovers, i now moved to a zip release. I will fix those settings

**Comment by gardiol** [comment]
Now, some pictures to show, some don't. They are all taken with my phone, so it does not make much sense.
here:
[https://www.gardiol.org/surya/broken.jpg](https://www.gardiol.org/surya/broken.jpg) does not show
[https://www.gardiol.org/surya/working.jpg](https://www.gardiol.org/surya/working.jpg) does show

**Comment by gardiol** [comment]
I found out (with some SQL searches) that the photo that does not show is present twice: one within the album and one with no album. After deleting from the GUI all the copies now i can upload it and display it properly? Does it make sense?

**Comment by gardiol** [comment]
Another question... the "big" folder is like the "original" folder but for the pictures loaded with the "ancient" release? Because my "original" folder is empty, except for the few i uploaded recently with 4.6.0-RC3.

**Comment by nagmat84** [comment]
> I found out (with some SQL searches) that the photo that does not show is present twice: one within the album and one with no album. After deleting from the GUI all the copies now i can upload it and display it properly? Does it make sense?

Partially, yes. The "black areas" usually appear, if Lychee believes that it has successfully created the size variants, but the actual image files for those size variants are missing. If you upload the same image a second time, Lychee detects that it is a duplicate and reuses the the already existing size variants to save some storage space. Of course, if the initially uploaded image is broken, so will the duplicate. Of course, if you delete all the instances of the same image and then re-upload the image, Lychee does not determine it to be a duplicate and creates the size variants anew.

In summary, your observation makes sense. However, the question is why Lychee failed to properly create size variants for the the initial picture which got you into this situation.

> Another question... the "big" folder is like the "original" folder but for the pictures loaded with the "ancient" release? Because my "original" folder is empty, except for the few i uploaded recently with 4.6.0-RC3.

Correct. With the new version the folders the high-dpi version of the size variants are stored in there own folders `thumb2x`, `small2x`, `medium2x` and the `big` folder has been renamed to `original`. However, the migration does not touch existing files.

**Comment by gardiol** [comment]
Ok, thanks you for your clarifications. Maybe the image failed to upload correctly the first time causing the issue? I managed to reload it just fine. I think the issue can be closed since i didnt managed to re produce it again.

**Comment by gardiol** [comment]
I had nothong in the logs about failing to create the smaller copies Anyway.
